src/main.cpp

```diff
@@ -2,23 +2,37 @@
 
 #include "acceleration.hpp"
 #include "barometer.hpp"
+#include "gps.hpp"
 
 constexpr auto baud_rate = 9600L;
+HardwareSerial Serial3(PC11, PC10);
 obc::Accelerometer acclrm;
 obc::Bmp bmp;
+obc::Gps gps(&Serial3);
+
+uint32_t timer = millis();
 
 void setup()
 {
     Serial.begin(baud_rate);
     Serial.println("setup");
     obc::init(acclrm);
     obc::init(bmp);
+    obc::init(gps);
 }
 
 void loop()
 {
+    constexpr int gps_interval = 2000;
     const auto [acclr, acclr_err] = obc::measure(acclrm);
     const auto [measurements, bmp_err] = obc::measure(bmp);
+    const auto [time, date, gpsposition, err] = obc::measure_gps(gps);
     if (acclr_err == obc::Error::Ok) { obc::print(acclr); }
     if (bmp_err == obc::Error::Ok) { obc::print(measurements); }
+    if (err == obc::Error::Ok && timer - millis() > gps_interval) {
```

Comment:
To powinno zostać rozbite. Error od gpsa jako oddzielny if, a timer jako główny if w loopie do triggera logów

Reply:
Nie jestem pewien czy to konieczne. A przynajmniej nie teraz.

Reply:
To na ten moment zmienną dam globalnie, a ten warunek zostanie tak jak jest. Najwyżej się zmieni w przyszłości

Reply:
if od timera będzie z automatu interwałem pomiarów dla wszystkich czujników. Odchodzi w ten sposób robienie dodatkowej zmiennej oddzielnej od timera z GPS (który i tak musi być) liczącej interwał. Dlatego jestem za tym, żeby mieć jedną zmienną odpowiadającą za trigger pomiarów.

Reply:
> Nie jestem pewien czy to konieczne. A przynajmniej nie teraz.

if od timera będzie z automatu interwałem pomiarów dla wszystkich czujników. Odchodzi w ten sposób robienie dodatkowej zmiennej oddzielnej od timera z GPS (który i tak musi być) liczącej interwał. Dlatego jestem za tym, żeby mieć jedną zmienną odpowiadającą za trigger pomiarów.